src/appleseed/renderer/kernel/lighting/directlightingintegrator.cpp

```diff
@@ -402,14 +402,20 @@ void DirectLightingIntegrator::add_emitting_triangle_sample_contribution(
 
     // Compute the transmission factor between the light sample and the shading point.
     Spectrum transmission;
-    m_material_sampler.trace_between(
-        m_shading_context,
-        sample.m_point,
-        transmission);
+    if(material->get_cast_shadows()){
```

Comment:
Please read and follow our coding conventions: https://github.com/appleseedhq/appleseed/wiki/Coding-Philosophy-and-Guidelines